realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -85,45 +91,66 @@ static RealmCacheType valueOf(Class<? extends BaseRealm> clazz) {
 
     // Realm path will be used as the key to store different RealmCaches. Different Realm configurations with same path
     // are not allowed and an exception will be thrown when trying to add it to the cache map.
-    private static final Map<String, RealmCache> cachesMap = new HashMap<>();
+    // A weak ref is used to hold the RealmCache instance. The weak ref entry will be cleared if and only if there
+    // is no Realm instance holding a strong ref to it and there is no Realm instance associated it is BEING created.
+    private static final Map<String, WeakReference<RealmCache>> cachesMap =
```

Comment:
The first version is not using the weak ref, and the `RealmCache` will not be cleared, instead, it will be re-used next time. 
I think the case you mentioned is more important than clearing unused cache from memory since we want to show the debug information about how many instances are not closed. I will revert the weak ref commit.